ts/features/mvl/__tests__/utils.test.ts

```diff
@@ -0,0 +1,91 @@
+import ReactNativeBlobUtil from "react-native-blob-util";
+
+import * as platform from "../../../utils/platform";
+import { mvlMockPdfAttachment } from "../types/__mock__/mvlMock";
+import { handleDownloadResult } from "../utils";
+
+jest.mock("../../../utils/platform");
+
+// eslint-disable-next-line functional/no-let
+let mockResponseStatus = 200;
+const defaultPath = mvlMockPdfAttachment.resourceUrl;
+
+// eslint-disable-next-line functional/immutable-data
+(ReactNativeBlobUtil as any).config = jest.fn(() => ({
+  fetch: jest.fn().mockReturnValue({
+    info: jest.fn().mockReturnValue({ status: mockResponseStatus }),
+    path: jest.fn().mockReturnValue(defaultPath)
+  })
+}));
+const mockAndroidUtil = jest.fn();
+// eslint-disable-next-line functional/immutable-data
+(ReactNativeBlobUtil as any).android = {
+  addCompleteDownload: mockAndroidUtil
+};
+const mockIosUtil = jest.fn();
+// eslint-disable-next-line functional/immutable-data
+(ReactNativeBlobUtil as any).ios = {
+  openDocument: mockIosUtil,
+  presentOptionsMenu: jest.fn()
+};
+
+const header = { auth: "It's me again!" };
+
+describe("handleDownloadResult utility function", () => {
+  beforeEach(() => {
+    mockAndroidUtil.mockReset();
+    mockIosUtil.mockReset();
+  });
+
+  it("should resolve successfully with an empty Promise", async () => {
+    await expect(
+      handleDownloadResult(mvlMockPdfAttachment, header)
+    ).resolves.toEqual(undefined);
+  });
+
+  describe("and there is an error", () => {
+    afterAll(() => {
+      mockResponseStatus = 200;
+    });
+    it("should reject with an Error instance", async () => {
+      mockResponseStatus = 500;
+      await expect(
+        handleDownloadResult(mvlMockPdfAttachment, header)
+      ).rejects.toEqual(
+        new Error("error 500 fetching htts://www.invoicepdf.com/invoce.pdf")
```

Comment:
```suggestion
        new Error("error 500 fetching " + mvlMockPdfAttachment.resourceUrl.href)
```
just to avoid domino effects when mock changes. Does it make sense?

Reply:
Makes sense. Fixed https://github.com/pagopa/io-app/pull/3706/commits/98709c6d6d3b566da81ae67ea8f87eef47d076bd#diff-aab3f362e2417890beb31e364e0dd9ec81d9c26b0a591c35828285c8301cd502L55